compiler/rustc_macros/src/query.rs

```diff
@@ -272,6 +272,14 @@ fn process_modifiers(query: &mut Query) -> QueryModifiers {
                 if desc.is_some() {
                     panic!("duplicate modifier `desc` for query `{}`", query.name);
                 }
+                if query.doc_comments.is_empty() {
+                    // FIXME: replace `{}` with `_` somehow
+                    let format_str = list.first().unwrap();
+                    let comment = parse_quote! {
+                        #[doc = #format_str]
+                    };
```

Comment:
Reposting here what I've mentioned on Zulip; feel free to edit / format as you see fit:

```suggestion
                    use ::syn::*;
                    let mut list = list.iter();
                    let format_str: String = match list.next() {
                        Some(&Expr::Lit(ExprLit { lit: Lit::Str(ref lit_str), .. })) => {
                            lit_str
                                .value()
                                .replace("`{}`", "{}") // We add them later anyways for consistency
                        },
                        _ => panic!("Expected a string literal"),
                    };
                    let mut fmt_fragments = format_str.split("{}");
                    let mut doc_string = fmt_fragments.next().unwrap().to_string();
                    list.map(::quote::ToTokens::to_token_stream)
                        .zip(fmt_fragments)
                        .for_each(|(tts, next_fmt_fragment)| {
                            use ::core::fmt::Write;
                            write!(
                                &mut doc_string,
                                " `{}` {}",
                                tts.to_string().replace(" . ", "."),
                                next_fmt_fragment,
                            ).unwrap();
                        })
                    ;
                    let comment = parse_quote! {
                        #[doc = #doc_string]
                    };
```

<details><summary>Preview</summary>

<img width="989" alt="Screen Shot 2020-12-24 at 18 00 33" src="https://user-images.githubusercontent.com/9920355/103101007-bb908400-4615-11eb-86a6-ac21c7b892dc.png">

</details>

Reply:
Added, thanks so much!